Student Resources/1 - Getting Started/2 - Constants, Variables, and Data Types/lab/Lab - Constants and Variables.playground/Contents.swift

```diff
@@ -32,19 +35,108 @@
  */
-
+var x = 1.1
+print(x)
+x = 2.2
+print(x)
+x = 3.3
+print(x)
+x = 4.4
+print(x)
 
 /*:
  Create a Boolean variable and set it to `true`. Print the variable, then assign it a value of `false`, and print it again.
  */
-
+var y = true
+print(y)
+y = false
+print(y)
 
 /*:
  Create two variables: one with a value of 0, the other with a value of 0.0. Try to assign the second variable to the first, and you'll receive an error. Add the necessary type annotation to allow the second variable to be assigned to the first.
  */
+var x1 = 0
+var x2 = 0.0
+x1 = Int(x2)
+
+/*:
+ Create a variable integer with a value of 1,000,000,000. Format it using commas, so it's easier to read.
+ */
+ var z = 1000000000
+ 
+import Foundation
+let numberFormatter = NumberFormatter()
+numberFormatter.numberStyle = .decimal
+print(numberFormatter.string(from: NSNumber(value:z))!)
+
+/*:
+ ## Exercise - Constants
+ 
+ Declare a constant called `friends` to represent the number of friends you have on social media. Give it a value between 50 and 1000. Print out the value by referencing your constant.
+ */
+let friends = 75
+
+/*:
+ Now assume you go through and remove a lot of your friends that aren't active on social media. Update your `friends` constant to a lower number than it currently is between 1 and 900.
+ */
+//friends = 50
+
+/*:
+ Does the above code compile? Why not? Print your explanation to the console using the `print` function. Go back and delete your line of code that updates the `friend` constant to a lower number so that the playground will compile properly.
+ */
+print("You can not change a value of a constant, switch to var to make it mutable")
 
+/*:
+ Declare a variable `age` and set it to your own age. Print `age` to the console.
+ */
+var age = 26
+print(age)
+
+/*:
+ Now pretend you just had a birthday, and update the `age` variable accordingly. Print `age` to the console.
+ */
+
+age = 27
+print(age)
+/*:
+ 
+ */
+
+
+/*:
+ Create a double variable with a value of 1.1. Update it to 2.2, 3.3, and 4.4. Print out the value after each assignment (again by referencing the variable you created).
+ */
+var x = 1.1
+print(x)
+x = 2.2
+print(x)
+x = 3.3
+print(x)
+x = 4.4
+print(x)
+
+/*:
+ Create a Boolean variable and set it to `true`. Print the variable, then assign it a value of `false`, and print it again.
+ */
+var y = true
+print(y)
+y = false
+print(y)
+
+/*:
+ Create two variables: one with a value of 0, the other with a value of 0.0. Try to assign the second variable to the first, and you'll receive an error. Add the necessary type annotation to allow the second variable to be assigned to the first.
+ */
+var x1 = 0
+var x2 = 0.0
+x1 = Int(x2)
 
 /*:
  Create a variable integer with a value of 1,000,000,000. Format it using commas, so it's easier to read.
  */
+ var z = 1000000000
```

Comment:
And you could even do something like this:
`var z = 1_000_000_000`
(The '_' characters get ignored by Swift but maybe make it easier to read what number it is we've written.